include/evmmax/evmmax.hpp

```diff
@@ -24,11 +35,18 @@ class ModArith
 public:
     explicit ModArith(const UintT& modulus) noexcept;
 
+    constexpr explicit ModArith(
+        const UintT& modulus, const UintT& r_squared, uint64_t modulus_inv) noexcept
+      : mod{modulus}, m_r_squared{r_squared}, m_mod_inv{modulus_inv}
+    {
+        // TODO: Add r_squared and modulus_inv values verification
+    }
```

Comment:
Yeah, please do.

Reply:
Discussed with @chfast . not so easy to do.